pkgs/development/python-modules/gpt4all-bindings/default.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  gpt4all,
+  gpt4all-cuda,
+  buildPythonPackage,
+  stdenv,
+  requests,
+  tqdm,
+  pythonOlder,
+  setuptools,
+  wheel,
+  python,
+  cuda ? false,
+}:
+let
+  basepkg = if cuda then gpt4all-cuda else gpt4all;
+
+  gpt4all-backend = stdenv.mkDerivation {
+    pname = "gpt4all-backend";
+    inherit (basepkg)
+      src
+      version
+      cmakeFlags
+      nativeBuildInputs
+      buildInputs
+      ;
+
+    sourceRoot = "${basepkg.src.name}/gpt4all-backend";
+    installPhase = ''
+      cp -r ./ $out
+    '';
+  };
+in
+buildPythonPackage {
+  pname = "gpt4all-bindings";
+  inherit (basepkg) src version;
+  sourceRoot = "${basepkg.src.name}/gpt4all-bindings/python";
+  pyproject = false;
+
+  patches = [ ./setup_fixup.patch ];
+  disabled = pythonOlder "3.12";
+
+  propagatedBuildInputs = [
+    requests
+    tqdm
+  ];
```

```suggestion
  dependencies = [
    requests
    tqdm
  ] ++ lib.optionals (pythonOlder "3.11") [
    typing-extension
  ];
```
https://github.com/nomic-ai/gpt4all/blob/a2b4529945460b200a226bc22f4861c24d285049/gpt4all-bindings/python/setup.py#L90-L95

---

Sorry I'll try to stick to the definitions in `setup.py` for my next Python packaging, instead of cooking using trial and error